lib/custom_typings/system.d.ts

```diff
@@ -0,0 +1,14 @@
+declare module 'system' {
```

Comment:
why do we need this?

Reply:
This addresses some typings issues that will come up when we do convert to typescript.